tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Azure/ServiceBus/ProcessMessageIntegration.cs

```diff
@@ -0,0 +1,102 @@
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(ProcessMessageIntegration));
+
+        /// <summary>
+        /// OnMethodBegin callback
+        /// </summary>
+        /// <typeparam name="TTarget">Type of the target</typeparam>
+        /// <typeparam name="TMessage">Type of the message argument</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method.</param>
+        /// <param name="message">Instance of the message</param>
+        /// <param name="cancellationToken">CancellationToken instance</param>
+        /// <returns>Calltarget state value</returns>
+        internal static CallTargetState OnMethodBegin<TTarget, TMessage>(TTarget instance, TMessage message, CancellationToken cancellationToken)
+            where TTarget : IReceiverManager
+            where TMessage : IServiceBusReceivedMessage
+        {
+            // Do not create a span, this will automatically be created by the Azure.Messaging.ServiceBus ActivitySource(s)
+            // when the following requirements are met:
+            // - AzureServiceBus integration enabled
+            // - DD_TRACE_OTEL_ENABLED=true
+            if (Tracer.Instance.Settings.IsIntegrationEnabled(IntegrationId.AzureServiceBus)
+                && Tracer.Instance.InternalActiveScope?.Span is var span)
+            {
+                PathwayContext? pathwayContext = null;
+
+                var tracer = Tracer.Instance;
+                var dataStreamsManager = tracer.TracerManager.DataStreamsManager;
+
+                if (dataStreamsManager.IsEnabled)
+                {
+                    if (message.ApplicationProperties is not null)
+                    {
+                        var headers = new ServiceBusHeadersCollectionAdapter(message.ApplicationProperties);
+
+                        try
+                        {
+                            pathwayContext = dataStreamsManager.ExtractPathwayContextAsBase64String(headers);
+                        }
+                        catch (Exception ex)
+                        {
+                            Log.Error(ex, "Error extracting PathwayContext from Azure Service Bus message");
+                        }
+                    }
+
+                    span.Context.MergePathwayContext(pathwayContext);
+
+                    var consumeTime = span.StartTime.UtcDateTime;
+                    var produceTime = message.EnqueuedTime.UtcDateTime;
+                    var messageQueueTimeMs = Math.Max(0, (consumeTime - produceTime).TotalMilliseconds);
+                    span.Tags.SetMetric(Trace.Metrics.MessageQueueTimeMs, messageQueueTimeMs);
+
+                    var namespaceString = instance.Processor.EntityPath;
+
+                    // TODO: we could pool these arrays to reduce allocations
+                    // NOTE: the tags must be sorted in alphabetical order
+                    var edgeTags = string.IsNullOrEmpty(namespaceString)
```

Comment:
As I mentioned in this [other comment](https://github.com/DataDog/dd-trace-dotnet/pull/4576/files#r1312381878), an incoming message can come directly from a queue or from a subscription (to a topic). Do we need to have a different tag for each type? The good thing is the subscription has a structure like `<topicName>/Subscriptions/<subscriptionName>` so if we need to distinguish between receiving from a queue and a subscription, it can be done.